cmake/CompilerWarnings.cmake

```diff
@@ -0,0 +1,80 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+function(set_default_compiler_warnings target_name)
+    option(WARNINGS_AS_ERRORS "Treat compiler warnings as errors" ON)
+
+    set(MSVC_WARNINGS
+        /W4 # Baseline reasonable warnings
```

Comment:
Recommendation would be to drop this to /W3.
Why? Even Microsoft headers don't compile cleanly with warning level 4.

Reply:
As per https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/185#discussion_r784115086, I removed the Windows warning config, leaving a warning message for when we come to compile on Windows, in 87bdfeda